### PR TITLE
Fix grouping of shipping rate by price and currency

### LINKED ISSUE #1189: Shipping rate input does not display the correct currency
### Describe the bug:

In Setup MC flow, the shipping rate input might not display the correct currency. This can happen when users setup a few shipping rates, then go to change the store currency in WooCommerce settings, and then go back to Setup MC to setup the remaining shipping rates. 

From the API call inspection, the shipping rate data is saved and retrieved with the correct currency value, it is just that the front-end is displaying the wrong currency.

image

📹 Demo video with my voice annotation:

https://user-images.githubusercontent.com/417342/149166106-4f9a3bf1-0a7f-4e68-ab15-fa5c7571d3fa.mov

### Expected behavior:

The input should display the correct currency.

### Actual behavior:

The input displays the wrong currency, which is based on the first shipping rate currency.

### Steps to reproduce:

1. Go to Setup MC, and proceed to setup a few shipping rates. 
2. Go to WooCommerce store settings and change the store currency.
3. Go back to Setup MC and proceed to setup a few more shipping rates. The new shipping rates would display the wrong currency.

### PR DESCRIPTION
**Pull request by ecgan**
### Changes proposed in this Pull Request:

Closes #1189.

Prior to this PR, in `getCountriesPriceArray` function, we were grouping shipping rates by price only, and for each of the grouping, the currency was set based on the currency of the first shipping rate. This causes issue #1189. 

In this PR, we fix the issue by grouping the shipping rates by price and currency.

Also, there were two copies of the `getCountriesPriceArray` function. I have refactored them and put the function into utils folder. 

Update: As per https://github.com/woocommerce/google-listings-and-ads/pull/1190#discussion_r783332727 below, I have also renamed the function `getCountriesPriceArray` to `groupShippingRatesByPriceCurrency`.

### Screenshots:

In Setup MC:

image

In Edit Free Listings:

image

### Detailed test instructions:

1. Go to Setup MC, and proceed to setup a few shipping rates.
2. Go to WooCommerce store settings and change the store currency.
3. Go back to Setup MC and proceed to setup a few more shipping rates. The new shipping rates should display the new store currency.
4. Reload the page. The shipping rates should be displayed with their associated currencies.
5. Complete the Setup MC flow.
6. Go to Dashboard tab and click on the edit button for Free Listings.
7. Go to Step 2. The shipping rates should be displayed with their associated currencies.

### Changelog entry

> Fix - Group shipping rate by price and currency, and display the right currency in shipping rate input.

### CODE REVIEW

File: js/src/utils/getCountriesPriceArray.js [hunk 1]
```diff
@@ -36,25 +41,30 @@
...
+ * //         countries: ['BR'],
+ * //         price: 20,
+ * //         currency: 'BRL',
+ * //     },
  * ]
  * ```
  *
  * @param {Array<Object>} shippingRates Array of shipping rates in the format of `{ countryCode, rate, currency }`.
  */
 const getCountriesPriceArray = ( shippingRates ) => {
```
Comment (jorgemd24) [inline js/src/utils/getCountriesPriceArray.js]: 💅  

The name of this file and function could be changed to groupCountriesByPriceCurrency or something similar. 
I think the current name does not describe well the function.  What do you think?

**Review by jorgemd24 — APPROVED** [review]
LGTM 👍 

💅  I left some comments about the function name.

What I have checked: 

✅  I have checked that the issue was present on the develop branch.
✅  I have checked that this PR is fixing the issue on the Setup MC onboarding.
✅  The issue has also been resolved on the Free Listings editing page.

File: js/src/utils/getCountriesPriceArray.js (hunk 1 above)
Comment (ecgan) [reply js/src/utils/getCountriesPriceArray.js]: @jorgemd24 , thanks for the fresh pair of eyes, you've got a good point! 😄 I'm thinking something like `groupShippingRatesByPriceCurrency`, will look into it. 👍

Comment (ecgan) [reply js/src/utils/getCountriesPriceArray.js]: I have renamed the function to `groupShippingRatesByPriceCurrency` in [`1c7d905` (#1190)](https://github.com/woocommerce/google-listings-and-ads/pull/1190/commits/1c7d905bdf59e20774126efec7bfa468e4357e7c).

**Pull request merged**